### PR TITLE
getting ready to publish to hashicorp registry

### PR DESCRIPTION
**Pull request by paulo-sampaio**

### CODE REVIEW

File: .goreleaser [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  name_template: '{{ .ProjectName }}_{{ .Version }}_{{ .Os }}_{{ .Arch }}'
+checksum:
+  name_template: '{{ .ProjectName }}_{{ .Version }}_SHA256SUMS'
+  algorithm: sha256
+signs:
+  - artifacts: checksum
+    args:
+      - "--batch"
+      - "--local-user"
+      - "{{ .Env.GPG_FINGERPRINT }}" # set this environment variable for your signing key
```
Comment (bonial-botuser) [inline .goreleaser:33]: I couldn't find this variable in 
https://github.com/crazy-max/ghaction-import-gpg

therefore  I think we should keep in our mind that we must add this in github variables

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+# or use an alternative one for key handling.
+#
+# You will need to pass the `--batch` flag to `gpg` in your signing step 
+# in `goreleaser` to indicate this is being used in a non-interactive mode.
+#
+name: release
+on:
+  push:
+    tags:
+      - 'v*'
```
Comment (sydorovdmytro) [inline .github/workflows/release.yml]: I would suggest `v[0-9].[0-9]+.[0-9]+`

File: .goreleaser [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    binary: terraform-provider-spinnaker
+    ldflags:
+    - -s -w
+    env:
+      - CGO_ENABLED=0
+    goos:
+      - linux
+      - darwin
+    goarch:
+      - '386'
```
Comment (sydorovdmytro) [inline .goreleaser]: not sure if we need this

File: .goreleaser [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+      - amd64
+    binary: '{{ .ProjectName }}_v{{ .Version }}'
+archives:
+  - format: binary
+    files:
+      - none*
+    replacements:
+      386: i386
```
Comment (sydorovdmytro) [inline .goreleaser]: not sure if we need this

File: .goreleaser [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+checksum:
+  name_template: 'checksums.txt'
+snapshot:
+  name_template: "{{ .Tag }}-next"
+changelog:
+  sort: asc
+  filters:
+    exclude:
+      - '^docs:'
+      - '^test:'
```
Comment (sydorovdmytro) [inline .goreleaser]: please add a new line

**Review by sydorovdmytro — CHANGES_REQUESTED** [review]

File: .goreleaser [hunk 6]
```diff
@@ -0,0 +1,52 @@
+---
```
Comment (martinohmann) [inline .goreleaser]: The file needs to be called `.goreleaser.yml`.

**Review by martinohmann — CHANGES_REQUESTED** [review]

**Comment by sydorovdmytro** [comment]
We should also change the readme to HCL2 syntax

**Comment by paulo-sampaio** [comment]
I think i have solved most of the issues, thank you all for the comments.

**Review by sydorovdmytro — APPROVED** [review]

**Review by martinohmann — APPROVED** [review]

**Pull request merged**